setuptools/command/upload_docs.py

```diff
@@ -59,7 +59,10 @@ def finalize_options(self):
         if self.upload_dir is None:
             if self.has_sphinx():
                 build_sphinx = self.get_finalized_command('build_sphinx')
-                self.target_dir = build_sphinx.builder_target_dir
+                for (builder, builder_dir) in build_sphinx.builder_target_dirs:
+                    if builder == "html":
+                        self.target_dir = builder_dir
+                        break
```

Comment:
Looks good. What happens if no builder == 'html'? Should we be concerned about that possibility?

Also, did you consider building a lookup map (dict) to resolve builder -> builder_dir? Were you aware you could construct a [dict from a sequence of pairs](https://docs.python.org/3/tutorial/datastructures.html#dictionaries), something like `dict(build_sphinx.builder_target_dirs)`?

Reply:
Hmm, that's true. The sphinx documentation says that the html builder is the default so I went with html. But if we want to let people upload any type, I think it would have to be an option in the command when they use `upload_docs` and then we can use whatever they inputted in place of where the `html` code is right now, or and do html by default if they run the command without any type option. Or we could keep as is, what do you think? 

Thanks for the idea, I'll rewrite it with dict!

Reply:
Sounds good. I also agree the case where "html" isn't present is an unlikely scenario and surely out of the scope of what to do here. I just wanted to think about that case. Happy to proceed with the current approach. Good reasoning.